framework/web/Response.php

```diff
@@ -1061,10 +1061,14 @@ protected function defaultFormatters()
      * Prepares for sending the response.
      * The default implementation will convert [[data]] into [[content]] and set headers accordingly.
      * @throws InvalidConfigException if the formatter for the specified format is invalid or [[format]] is not supported
+     *
+     * @see https://tools.ietf.org/html/rfc7231#page-53
+     * @see https://tools.ietf.org/html/rfc7232#page-18
      */
     protected function prepare()
     {
-        if ($this->statusCode === 204) {
+        if ($this->statusCode === 204 || $this->statusCode === 304) {
```

Comment:
better use `$this->getStatusCode()`

Reply:
Also I've changed the condition expression, to reduce the cognitive load